syntax/scan.go

```diff
@@ -227,8 +227,14 @@ func readSource(filename string, src interface{}) (data []byte, err error) {
 	case string:
 		data = []byte(src)
 	case []byte:
+		if src == nil {
+			return readSource(filename, nil)
+		}
 		data = src
 	case io.Reader:
+		if src == nil {
```

Comment:
This condition is provably false, because an interface-typed case in a type switch can never be reached with a nil value.